core/chains/evm/logpoller/parser.go

```diff
@@ -268,15 +268,14 @@ func (v *pgDSLParser) whereClause(expressions []query.Expression, limiter query.
 			return "", errors.New("invalid cursor direction")
 		}
 
-		block, txHash, logIdx, err := valuesFromCursor(limiter.Limit.Cursor)
```

Comment:
in case of reorg this is not an unique event identifier, txHash_logIdx pair is the only evm unique identifier

Reply:
That is true, good callout. If we leave `txHash` in the cursor, it wouldn't alter the order in any way for log order since we still have `logIdx`. In that case, we won't need to make any changes at all.

Reply:
> in case of reorg this is not an unique event identifier, txHash_logIdx pair is the only evm unique identifier

Maybe I should look at the surrounding code, but are you sure this is something we need?  LogPoller always returns blocks from the current canonical chain, it can never return more than one block with the same block number in a given query. In any given query, block number + log index uniquely identifies each log and uniquely specifies the order of them.  (In fact, it's not even possible to store multiple rows in the db for the same (block_number, log_index) combination)... those are always guaranteed to be unique.  The only case where it might be something you need is if something wants to get old blocks that are re-org'd out and keep track of both them and the canonical blocks separately.  But usually I would think you'd *want* them to be identified with the same key so that you can replace the old one with the current one.  (Why would a product ever care about old blocks that have been re-org'd out, unless it's just to replace them?)

Reply:
Even if log poller handles reorgs this is more precise and helpful if products need to log this value, I don't see a reason not to do this

Reply:
> If we leave txHash in the cursor, it wouldn't alter the order in any way for log order since we still have logIdx

Depends what order you have those listed in.  If it's the order in the return values above (block, txHash, logIdx) then leaving txHash in there will definitely destroy the ordering.  The transaction hashes are random, so it will make everything in random order instead of being in order of how the logs were emitted on chain.  If logIdx is listed last than it will not have any effect since it's already ordered everything in the wrong order.

Reply:
> Even if log poller handles reorgs this is more precise and helpful if products need to log this value, I don't see a reason not to do this

Don't they already get that field back with each log?

Reply:
Cursor extracted from the log event should have txHash and since that is the same cursor getting passed into here they should be the same

Reply:
Resolved by moving `txHash` to the end of the cursor and limiting cursor usage to finalized.